### PR TITLE
Add datasets APIs into CLI

### PR DESCRIPTION
**Pull request by harshbhatt7585**
We want to do the followings:
1. Create functions which helps us to call Datasets APIs
2. Create CLI flags which help us to directly access Datasets functions

### CODE REVIEW

**Review by smellslikeml — COMMENTED** [review]
looks good :+1:  
@salma-remyx is testing it out

File: remyxai/api/datasets.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    """Download dataset by generating a presigned URL."""
+    url = f"{BASE_URL}/datasets/download/{dataset_type}/{dataset_name}"
+    response = requests.get(url, headers=HEADERS, stream=True)
+
+    log_api_response(response)
+
+    if response.status_code == 200:
+        presigned_url = response.json().get("presigned_url", "")
+        if presigned_url:
+            filename = f"{dataset_name}.zip"
```
Comment (salma-remyx) [inline remyxai/api/datasets.py:31]:
```suggestion
            filename = f"{dataset_name}.csv"
```

**Review by salma-remyx — CHANGES_REQUESTED** [review]
updated the download to expect a `.csv` file instead of a zip and updating the `run_datacomposer` function to send a file using the correct headers

File: remyxai/api/tasks.py [hunk 2]
```diff
@@ -86,3 +86,32 @@ def train_generator(model_name: str, hf_dataset: str):
     params = {"hf_dataset": hf_dataset}
     response = requests.post(url, headers=HEADERS, params=params)
     return response.json()
+
```
Comment (salma-remyx) [inline remyxai/api/tasks.py:89]: Also, move this to the top at the imports but add 
```
from typing import Optional
```
for the suggested changes to work

Comment (harshbhatt7585) [reply remyxai/api/tasks.py:91]: Added new commit :)

**Review by salma-remyx — APPROVED** [review]

**Pull request merged**